xwiki-rendering-macros/xwiki-rendering-macro-html/src/main/java/org/xwiki/rendering/internal/macro/html/HTMLMacro.java

```diff
@@ -152,83 +139,28 @@ public List<Block> execute(HTMLMacroParameters parameters, String content, Macro
                 normalizedContent = renderWikiSyntax(normalizedContent, context.getTransformation(), context);
             }
 
-            // Clean the HTML into valid XHTML if the user has asked (it's the default).
-            if (parameters.getClean()) {
-                normalizedContent = cleanHTML(normalizedContent, context);
-            } else if (context.getTransformationContext().isRestricted()) {
-                throw new MacroExecutionException(
-                    "The HTML macro may not be used with clean=\"false\" in this context.");
+            RawBlock contentBlock = new RawBlock(normalizedContent, targetSyntax);
+
+            try {
+                RawBlockFilterParameters filterParameters = new RawBlockFilterParameters(context);
+                filterParameters.setClean(parameters.getClean());
+                List<RawBlockFilter> filters = this.componentManager.getInstanceList(RawBlockFilter.class);
+                filters.sort((filter1, filter2) -> filter1.getPriority() - filter2.getPriority());
```

Comment:
In my opinion, `Comparator.comparingInt` would be more readable here:
```suggestion
                filters.sort(Comparator.comparingInt(RawBlockFilter::getPriority));
```

Reply:
Thanks, didn't know about it.